src/rest/SearchEndpoint.ts

```diff
@@ -873,4 +821,404 @@ export class SearchEndpoint implements ISearchEndpoint {
+          requestData: data,
+          method: '',
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Set the request data type
+ * @param type The type to set
+ */
+function requestDataType(type: string){
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    let originalMethod = descriptor.value;
+    let nbParams = target[key].prototype.constructor.length;
+
+    descriptor.value = function(...args: any[]) {
+      if(args[nbParams-1]){
+        args[nbParams-1].requestDataType = type;
+      }
+      else{
+        let params: IEndpointCallParameters = {
+          url: '',
+          queryString: [],
+          requestData: {},
+          requestDataType: type,
+          method: '',
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Set the request data type
+ * @param met The type to set
+ */
+function method(met: string){
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    let originalMethod = descriptor.value;
+    let nbParams = target[key].prototype.constructor.length;
+
+    descriptor.value = function(...args: any[]) {
+      if(args[nbParams-1]){
+        args[nbParams-1].method = met;
+      }
+      else{
+        let params: IEndpointCallParameters = {
+          url: '',
+          queryString: [],
+          requestData: {},
+          method: met,
+          responseType: '',
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Set the response type
+ * @param resp The response type to set
+ */
+function responseType(resp: string){
+  return function(target: Object, key: string, descriptor: TypedPropertyDescriptor<any>) {
+    let originalMethod = descriptor.value;
+    let nbParams = target[key].prototype.constructor.length;
+
+    descriptor.value = function(...args: any[]) {
+      if(args[nbParams-1]){
+        args[nbParams-1].responseType = resp;
+      }
+      else{
+        let params: IEndpointCallParameters = {
+          url: '',
+          queryString: [],
+          requestData: {},
+          method: '',
+          responseType: resp,
+          errorsAsSuccess: false
+        };
+        args[nbParams-1] = params;
+      }
+      let result = originalMethod.apply(this, args);
+      return result;
+    };
+
+    return descriptor;
+  }
+}
+
+/**
+ * Set the errorsAsSuccess parameter
+ * @param error The value to set
+ */
+function errorsAsSuccess(error: boolean){

```

Comment:
I don't think the param is necessary : When this decorator is used, it means the method will always want to set `true`